Helper/Data.php

```diff
@@ -228,4 +237,44 @@ protected function make_request($path, $params) {
         $response = file_get_contents($url);
         return $response == '1';
     }
-}
\ No newline at end of file
+
+    /**
+     * @param string $path
+     * @param array $params
+     * @param string $method
+     * @return bool|string
+     * @throws \Exception
+     */
+    private function sendApiRequest(string $path, array $params, string $method = 'POST')
+    {
+        $url = self::KLAVIYO_HOST . $path;
+
+        $curl = curl_init();
+        $encodedParams = json_encode($params);
+
+        curl_setopt_array($curl, [
+            CURLOPT_URL => $url,
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_CUSTOMREQUEST => $method,
+            CURLOPT_POSTFIELDS => $encodedParams,
+            CURLOPT_USERAGENT => self::USER_AGENT,
+            CURLOPT_HTTPHEADER => [
+                'Content-Type: application/json',
+                'Content-Length: ' . strlen($encodedParams)
+            ],
+        ]);
+
+        // Submit the POST request
```

Comment:
could we remove this comment or just make it:
```suggestion
        // Submit the request
```

Reply:
Sure

Reply:
Done